library/src/main/java/org/xmtp/android/library/SigningKey.kt

```diff
@@ -13,15 +13,16 @@ import org.xmtp.android.library.messages.rawData
 import org.xmtp.proto.message.contents.PrivateKeyOuterClass
 import org.xmtp.proto.message.contents.PublicKeyOuterClass
 import org.xmtp.proto.message.contents.SignatureOuterClass
+import uniffi.xmtpv3.FfiInboxOwner
 import java.math.BigInteger
 import java.util.Date
 
-interface SigningKey {
-    val address: String
-
+interface SigningKey: FfiInboxOwner {
     suspend fun sign(data: ByteArray): SignatureOuterClass.Signature?
 
-    suspend fun sign(message: String): SignatureOuterClass.Signature?
+    suspend fun signLegacy(message: String): SignatureOuterClass.Signature?
+
+    override fun sign(text: String): ByteArray
```

Comment:
Java doesn't allow name shadowing since they both take a String even if they have different return values. To make this change smaller I just renamed the existing sign to `signLegacy` for now but hopefully we can remove it entirely.